src/backend/core/models.py

```diff
@@ -284,8 +284,31 @@ def __str__(self):
         return f"{id_str:s}{main_str:s}"
 
     def save(self, *args, **kwargs):
-        """Ensure users always have one and only one main identity."""
+        """
+        Saves identity, ensuring users always have exactly one main identity.
+        Also converts valid invitations to accesses upon creating an identity.
+        """
+
+        # If new identity, convert all valid invitations to team accesses.
+        # Expired invitations are ignored.
+        if self._state.adding:
+            validity_duration = timedelta(seconds=settings.INVITATION_VALIDITY_DURATION)
+            active_invitations = Invitation.objects.filter(
+                email=self.email, created_at__gte=timezone.now() - validity_duration
+            )
+            TeamAccess.objects.bulk_create(
+                [
+                    TeamAccess(
+                        user=self.user, team=invitation.team, role=invitation.role
+                    )
+                    for invitation in active_invitations
+                ]
+            )
+            active_invitations.delete()
```

Comment:
niceee, have you inspected the number of SQL queries, to make sure only two are triggered now?

Reply:
Unfortunately, django_assert_num_queries reports 16 queries + 1 for each invitation :/  

`.bulk_create()` and `.delete()` are only one query each, so that's nice.
But the comprehensive list inside the `bulk_create()` results in this query being repeated for each invitation. I'm not sure I understand.

```
SELECT "people_team"."id", "people_team"."created_at", "people_team"."updated_at", "people_team"."name", "people_team"."slug" FROM "people_team" WHERE "people_team"."id" = '2c5e952e221f47d59cfadf079295c56a'::uuid LIMIT 21
```

Reply:
oh my god, is this a select_related situation ? :o

Reply:
OMG, it was. 💡🧠💡 !!!

Reply:
nice investigation 🕵️ I would test it